comp/otelcol/collector/component.go

```diff
@@ -23,14 +24,16 @@ import (
 // of the logs agent component.
 
 // Component specifies the interface implemented by the collector module.
-type Component interface {
-	Start() error
-	Stop()
-	Status() otlp.CollectorStatus
-}
+type Component = collectordef.Component
 
 // Module specifies the Collector module bundle.
 func Module() fxutil.Module {
 	return fxutil.Component(
 		fx.Provide(newPipeline))
```

Comment:
I forsee an fx problem this is going to cause:

1) newPipeline provides to fx: `collector.Component` and `status.InformationProvider`
2) This PR is adding a provider for `collector.Component`

It's an error in fx for more than 1 provider to yield the same type, unless they are a value group. Is the old `newPipeline` constructor going away? Or do we want there to be more than 1 collector, in which case we need to introduce a value group.